lib/ain-rs-exports/src/evm.rs

```diff
@@ -1187,16 +1187,43 @@ pub fn evm_try_get_tx_sender_info_from_raw_tx(
         return cross_boundary_error_return(result, "prepay fee value overflow");
     };
 
+    let Ok(TxResponse { used_gas, .. }) = SERVICES.evm.core.call(EthCallArgs {
```

Comment:
Note that the gas used calculated here is based on the state of the current tip, which means that we cannot make the assumption that this gas used is actually correct. The only pipeline where we will be able to correctly get the gas used calculation is only on validate evm tx pipeline.

Reply:
We do execute the TX to get the gas, it should be approximately correct. 

Problem we were having is the miners becoming unresponsive while creating blocks due to lots of large fee TXs in the mempool. The mempool was spammed with thousands of 10m fee TXs which took down the miner nodes. This PR prevents those TXs from being processed after the block is full, it does not need to be perfect.

Currently lots of 10M fee TXs act like a DoS on staking nodes.

Reply:
`We do execute the TX to get the gas, it should be approximately correct.`
This might not be true. A evm tx that executes with an early return could give a vastly different gas used from an evm tx that executes fully.

Reply:
But the assumption should be good enough to protect the minter from DoS attack, while trading off some performance.